### PR TITLE
minMaxSum problem for code review

### PR DESCRIPTION
**Pull request by kpacholak**

### CODE REVIEW

File: minMaxSum.swift [hunk 1]
```diff
@@ -1 +1,39 @@
 
+// Mini-Max Sum algorithm problem
+// https://www.hackerrank.com/challenges/mini-max-sum/problem
+// Swift solution
+// Krzysztof Pacholak
+
+
+let arr = [3, 2, 1, 4, 5]
+
+func znajdzMinMaxSume(_ arr: [Int]) {
```
Comment (maciekzieba) [inline minMaxSum.swift:10]: Please use only english names (for vars and functions)

File: minMaxSum.swift [hunk 2]
```diff
@@ -1 +1,39 @@
...
+        if i < tymczasowyMin {
+            tymczasowyMin = i
+        } else if i > tymczasowyMax {
+            tymczasowyMax = i
+        }
+    }
+    
+    var suma = 0
+    
+    for j in 0..<arr.count {
```
Comment (maciekzieba) [inline minMaxSum.swift:27]: why do you need another loop for sum? Can't you do it in the previous one?

**Review by maciekzieba — CHANGES_REQUESTED** [review]

File: minMaxSum.swift (hunk 1 above)
Comment (maciekzieba) [inline minMaxSum.swift:10]: Can you also provide unit tests for this function?

File: minMaxSum.swift (hunk 2 above)
Comment (kpacholak) [reply minMaxSum.swift:27]: ok, got it. So here is my new loop

```
 for i in arr {
        sum += arr[i-1]
        if i < currentMin {
            currentMin = i
        } else if i > currentMax {
            currentMax = i
        }
    }
```

File: minMaxSum.swift (hunk 1 above)
Comment (kpacholak) [reply minMaxSum.swift:10]: no idea 🤷‍♂️ but I will work on it

File: minMaxSum.swift (hunk 2 above)
Comment (maciekzieba) [reply minMaxSum.swift:27]: Just push the new code to this branch

File: minMaxSum.swift [hunk 3]
```diff
@@ -1 +1,36 @@
...
+func findMinMaxSum(_ arr: [Int]) {
+    
+    var currentMax = arr[0]
+    var currentMin = arr[0]
+    var minSum = 0
+    var maxSum = 0
+    var sum = 0
+    
+    for i in arr {
+        sum += arr[i-1]
```
Comment (maciekzieba) [inline minMaxSum.swift:18]: I do not think this will work... remember that i represents values over arr not indexes...

Comment (kpacholak) [reply minMaxSum.swift:18]: ok, I see. We needed an index variable here :) 

```
 var currentMax = arr[0]
    var currentMin = arr[0]
    var minSum = 0
    var maxSum = 0
    var sum = 0
    var index = 0
    
    for i in arr {
        sum += arr[index]
        if i < currentMin {
            currentMin = i
        } else if i > currentMax {
            currentMax = i
        }
        index += 1
    }
    

    
    minSum = sum-currentMax
    maxSum = sum-currentMin
    
    print(minSum, maxSum)
```

File: minMaxSum.swift [hunk 4]
```diff
@@ -1 +1,38 @@
...
+    
+    var currentMax = arr[0]
+    var currentMin = arr[0]
+    var minSum = 0
+    var maxSum = 0
+    var sum = 0
+    var index = 0
+    
+    for i in arr {
+        sum += arr[index]
```
Comment (maciekzieba) [inline minMaxSum.swift:19]: Why not sum += i ?

**Pull request merged**